org.goldendict.desktop.yml

```diff
@@ -0,0 +1,130 @@
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications
+rename-appdata-file: goldendict.appdata.xml
+rename-desktop-file: goldendict.desktop
+modules:
+  - name: opencc
+    buildsystem: cmake-ninja
+    build-options:
+      env:
+        CC: 'ccache gcc'
+        CXX: 'ccache g++'
+    sources:
+      - type: git
+        url: https://github.com/BYVoid/OpenCC
+        tag: ver.1.1.1
+        x-checker-data:
+          type: anitya
+          project-id: 7230
+          tag-template: ver.$version
+
+  - name: lzo2
+    config-opts:
+      - '--enable-shared'
+      - '--disable-static'
+    cleanup:
+      - /include
+      - /share/doc
+      - '*.la'
+      - /lib/*.so
+    sources:
+      - type: archive
+        url: 'https://www.oberhumer.com/opensource/lzo/download/lzo-2.10.tar.gz'
+        sha256: c0f892943208266f9b6543b3ae308fab6284c5c90e627931446fb49b4221a072
+        x-checker-data:
+          type: anitya
+          project-id: 1868
+          url-template: http://www.oberhumer.com/opensource/lzo/download/lzo-$version.tar.gz
+
+  - name: eb
+    config-opts:
+      - '--enable-shared'
+      - '--disable-static'
+    cleanup:
+      - /bin
+      - /share/doc
+    sources:
+      - type: git
+        url: https://salsa.debian.org/debian/eb
+        tag: debian/4.4.3-13
+        x-checker-data:
+          type: anitya
+          project-id: 155542
+          tag-template: debian/$version
+
+  - name: zstd
+    'buildsystem': 'meson'
+    'subdir': 'build/meson'
+    'config-opts':
+      - -Dbin_programs=false
+      - -Dbin_contrib=false
+    sources:
+      - type: git
+        url: 'https://github.com/facebook/zstd.git'
+        tag: v1.4.8
+        x-checker-data:
+          type: anitya
+          project-id: 12083
+          tag-template: v$version
+
+  - name: libao
+    cleanup:
+      - /include
+      - '*.la'
+      - /lib/pkgconfig
+      - /lib/ckport
+      - /share
+    sources:
+      - type: archive
+        url: 'https://downloads.xiph.org/releases/ao/libao-1.2.0.tar.gz'
+        sha256: 03ad231ad1f9d64b52474392d63c31197b0bc7bd416e58b1c10a329a5ed89caf
+        x-checker-data:
+          type: anitya
+          project-id: 7629
+          url-template: 'https://downloads.xiph.org/releases/ao/libao-$version.tar.gz'
+
+  - name: goldendict
+    buildsystem: qmake
+    build-options:
+      env:
+        CC: 'ccache gcc'
+        CXX: 'ccache g++'
+        QMAKEPATH: '/app/lib'
+        LD_LIBRARY_PATH: ${LD_LIBRARY_PATH}:/app/lib/:/usr/lib
+    config-opts:
+      - DEFINES+=USE_DBUS
+      - INCLUDEPATH+="${FLATPAK_DEST}/include/QtWebKitWidgets"
+      - INCLUDEPATH+="${FLATPAK_DEST}/include/QtWebKit"
+      - CONFIG+=release
+    build-commands:
+      - sed -i "s,Icon=goldendict,Icon=${FLATPAK_ID}," redist/goldendict.desktop
+      - sed -i "s,<launchable type="desktop-id">goldendict.desktop</launchable>,<launchable type="desktop-id">${FLATPAK_ID}.desktop</launchable>," redist/goldendict.appdata.xml
```

Comment:
Do that as a separate patch & send it upstream if please.

Reply:
That just renames `goldendict` to its flatpak id. Obviously won't be in the upstream. Using three (patch, desktop-file-edit, appstream-util) looks a little messy. But of course that could be fixed in a better way if  appstream-util could edit the metadata more eligantly

Reply:
Why it won't be in the upstream?

Reply:
Because usually the `Icon=` is like `goldendict` rather than `org.goldendict.desktop`? I can't find a good reason to persuade the upstream. They would say like "it's the flatpak-only format".

Reply:
It's not the Flatpak-only format. see https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic & https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming.

Reply:
Great to know. That's a good backup since the specs say so. I'll have a try. But could it be just a link to the PR before the upstream accepts it? The PR may have any destinies anyway.